dhis-2/dhis-e2e-test/src/test/java/org/hisp/dhis/tracker/importer/IdSchemeTests.java

```diff
@@ -0,0 +1,242 @@
+            .array();
+
+        trackerActions.postAndGetJobReport( payload,
+            new QueryParamsBuilder()
+                .addAll( "async=false", "idScheme=" + idScheme ) )
+            .validateSuccessfulImport();
+    }
+
+    @MethodSource( "provideIdSchemeArguments" )
+    @ParameterizedTest( name = "POST to /tracker?idScheme={0}" )
+    public void shouldImportEventProgramDataByIdScheme( String idScheme )
+    {
+        TestData data = new TestData( idScheme );
+
+        JsonObject object = new EventDataBuilder()
+            .setOu( data.getOrgUnit() )
+            .setProgramStage( data.getEventProgramStage() )
+            .setProgram( data.getEventProgram() )
+            .setAttributeCategoryOptions( Arrays.asList( data.categoryOption ) )
+            .array();
+
+        trackerActions
+            .postAndGetJobReport( object, new QueryParamsBuilder().addAll( "async=false", "idScheme=" + idScheme ) )
+            .validateSuccessfulImport();
+    }
+
+    @Test
+    public void shouldImportEventsWithDifferentProgramSchemes()
+    {
+        String programIdScheme = "CODE";
+        String ouIdScheme = "NAME";
+        String programStageIdScheme = "ATTRIBUTE:" + ATTRIBUTE_ID;
+        TestData data = new TestData();
+
+        JsonObject event = new EventDataBuilder()
+            .setProgram( new TestData( programIdScheme ).getEventProgram() )
+            .setProgramStage( new TestData( programStageIdScheme ).getEventProgramStage() )
+            .setOu( new TestData( ouIdScheme ).getOrgUnit() )
+            .setAttributeCategoryOptions( Arrays.asList( data.getCategoryOption() ) )
+            .array();
+
+        // act
+        trackerActions
+            .postAndGetJobReport( event, new QueryParamsBuilder()
+                .add( "programIdScheme", programIdScheme )
+                .add( "orgUnitIdScheme", ouIdScheme )
+                .add( "programStageIdScheme", programStageIdScheme ) )
+            .validateSuccessfulImport();
+    }
+
+    @Getter
+    @Setter
+    private static class TestData
+    {
+        private String trackedEntityType = "mthkj6qr5y9";
+
+        private String orgUnit = "yMXcwGmzIWY";
+
+        private String trackedEntityAttribute = "Kg6I0Cl3C7r";
+
+        private String trackerProgram = "iAI6kmFqoOc";
+
+        private String trackerProgramStage = "eTaBehVASzG";
+
+        private String dataElement = "VkoGQvbzHk2";
+
+        private String eventProgram = "jDnjGYZFkA4";
+
+        private String eventProgramStage = "fFNTQZPt2J4";
+
+        private String categoryOption = "fjvZIRlTBrp";
+
+        private String relationshipType = "XDaaLiqMYKy";
+
+        public TestData()
+        {
+        }
+
+        public TestData( String idScheme )
+        {
+            String propertyName;
+            if ( idScheme.toLowerCase().contains( "attribute" ) )
+            {
+                propertyName = "attributeValues.value[0]";
+            }
```

Comment:
```suggestion
            } else {
```

the newline in between if and else looks odd